Add Hyperdrive example with Nile database (#18374)
cloudflare/cloudflare-docs: src/content/docs/hyperdrive/examples/nile.mdx

---
type: example
summary: Connect Hyperdrive to a Nile database instance.
pcx_content_type: example
title: Connect to Nile
sidebar:
  order: 3
description: Connect Hyperdrive to a Nile database instance.
---

import { Render } from "~/components";

This example shows you how to connect Hyperdrive to a [Nile](https://thenile.dev) PostgreSQL database instance.

> Nile is PostgreSQL re-engineered for multi-tenant applications. Nile's virtual tenant databases provides you with isolation, 
> placement, insights and more for your tenant's data and embedding. You can learn more in [Nile documentation](https://www.thenile.dev/docs/getting-started/whatisnile).

Reviewer: ```suggestion
Nile is PostgreSQL re-engineered for multi-tenant applications. Nile's virtual tenant databases provide you with isolation, placement, insight, and other features for your tenant's data and embedding. Refer to [Nile documentation](https://www.thenile.dev/docs/getting-started/whatisnile) to learn more.
```

## 1. Allow Hyperdrive access

> You can connect Cloudflare Hyperdrive to any Nile database in your workspace using its connection string - either with new set of 
> credentials or using existing set.

Reviewer: ```suggestion
You can connect Cloudflare Hyperdrive to any Nile database in your workspace using its connection string - either with a new set of credentials, or using an existing set.
```

> ### Nile Console

Reviewer: ```suggestion
### Nile console
```

To get a connection string from Nile Console:

1. Log in to [Nile Console](https://console.thenile.dev) and select a database 
2. On the left hand menu, click on "Settings" (the bottom-most icon) and then select "Connection"

> 3. Click on the elephant, and you'll see a connection string show up. 
> 4. If you want a new pair of credentials for use with Cloudflare Hyperdrive (recommended!), click on "Generate credentials".
> 5. Copy the connection string without the "psql" part

Reviewer 2: ```suggestion
3. Select the PostgreSQL logo to show the connection string. 
4. Select "Generate credentials" to generate new credentials.
5. Copy the connection string (without the "psql" part).
```

> To get a connection string from Nile Console:
> 
> 1. Log in to [Nile Console](https://console.thenile.dev) and select a database 
> 2. On the left hand menu, click on "Settings" (the bottom-most icon) and then select "Connection"
> 3. Click on the elephant, and you'll see a connection string show up. 
> 4. If you want a new pair of credentials for use with Cloudflare Hyperdrive (recommended!), click on "Generate credentials".
> 5. Copy the connection string without the "psql" part
> 
> You'll end up with a connection string similar to this:

Reviewer: ```suggestion
To get a connection string from Nile console:

1. Log in to [Nile console](https://console.thenile.dev), then select a database.
2. On the left hand menu, click **Settings** (the bottom-most icon) and then select **Connection**.
3. Click on the elephant, and you will see a connection string.
4. If you want a new pair of credentials to use with Cloudflare Hyperdrive (recommended), click **Generate credentials**.
5. Copy the connection string without the "psql" part.

You will have obtained a connection string similar to the following:
```
